helpdesk/teams/views.py

```diff
@@ -172,3 +174,38 @@ def get_entries(self) -> QuerySet[Any]:
 
     def get_context_data(self, **kwargs: Any) -> dict[str, Any]:
         return super().get_context_data(entries=self.get_entries(), **kwargs)
+
+
+class TeamAttendanceView(LoginRequiredMixin, SingleTableMixin, ListView):
+    model = Team
+    table_class = TeamAttendanceTable
+
+    def get_queryset(self) -> QuerySet[Any]:
+        return Team.objects.all().prefetch_related(
+            Prefetch(
+                "team_attendance_events",
+                TeamAttendanceEvent.objects.order_by("-created_at")[:1],
+                to_attr="latest_event",
+            )
+        )
+
+
+class TeamAttendanceFormView(LoginRequiredMixin, CreateView):
+    model = TeamAttendanceEvent
+    form_class = TeamAttendanceLogForm
+    slug_field = "tla"
+
+    def get_success_url(self) -> str:
+        return reverse_lazy("teams:team_list_attendance")
+
+    def get_initial(self) -> dict[str, Any]:
+        return {"team": get_object_or_404(Team, tla=self.kwargs["slug"])}
+
+    def get_context_data(self, **kwargs: Any) -> dict[str, Any]:
+        context_data = super().get_context_data(**kwargs)
+        context_data["team"] = self.kwargs["slug"]
```

Comment:
I _think_ you can get this from `context_data["form"].initial["team"]` instead, and that'll give you the entire team object, rather than just the TLA.